logger/src/main/java/com/orhanobut/logger/LoggerPrinter.java

```diff
@@ -0,0 +1,372 @@
+
+    /**
+     * Android's max limit for a log entry is ~4076 bytes,
+     * so 4000 bytes is used as chunk size since default charset
+     * is UTF-8
+     */
+    private static final int CHUNK_SIZE = 4000;
+
+    /**
+     * It is used for json pretty print
+     */
+    private static final int JSON_INDENT = 4;
+
+    /**
+     * The minimum stack trace index, starts at this class after two native calls.
+     */
+    private static final int MIN_STACK_OFFSET = 3;
+
+    /**
+     * It is used to determine log settings such as method count, thread info visibility
+     */
+    private static final Settings settings = new Settings();
+
+    /**
+     * Drawing toolbox
+     */
+    private static final char TOP_LEFT_CORNER = '╔';
+    private static final char BOTTOM_LEFT_CORNER = '╚';
+    private static final char MIDDLE_CORNER = '╟';
+    private static final char HORIZONTAL_DOUBLE_LINE = '║';
+    private static final String DOUBLE_DIVIDER = "════════════════════════════════════════════";
+    private static final String SINGLE_DIVIDER = "────────────────────────────────────────────";
+    private static final String TOP_BORDER = TOP_LEFT_CORNER + DOUBLE_DIVIDER + DOUBLE_DIVIDER;
+    private static final String BOTTOM_BORDER = BOTTOM_LEFT_CORNER + DOUBLE_DIVIDER + DOUBLE_DIVIDER;
+    private static final String MIDDLE_BORDER = MIDDLE_CORNER + SINGLE_DIVIDER + SINGLE_DIVIDER;
+
+    /**
+     * TAG is used for the Log, the name is a little different
+     * in order to differentiate the logs easily with the filter
+     */
+    private static String TAG = "PRETTYLOGGER";
+
+    /**
+     * Localize single tag and method count for each thread
+     */
+    private static final ThreadLocal<String> LOCAL_TAG = new ThreadLocal<>();
+    private static final ThreadLocal<Integer> LOCAL_METHOD_COUNT = new ThreadLocal<>();
+
+    /**
+     * It is used to change the tag
+     *
+     * @param tag is the given string which will be used in Logger
+     */
+    @Override
+    public Settings init(String tag) {
+        if (tag == null) {
+            throw new NullPointerException("tag may not be null");
+        }
+        if (tag.trim().length() == 0) {
+            throw new IllegalStateException("tag may not be empty");
+        }
+        LoggerPrinter.TAG = tag;
+        return settings;
+    }
+
+    @Override
+    public Printer t(String tag, int methodCount) {
+        if (tag != null) {
+            LOCAL_TAG.set(tag);
+        }
+        LOCAL_METHOD_COUNT.set(methodCount);
+        return this;
+    }
+
+    @Override
+    public void d(String message, Object... args) {
+        log(Log.DEBUG, message, args);
+    }
+
+    @Override
+    public void e(String message, Object... args) {
+        e(null, message, args);
+    }
+
+    @Override
+    public void e(Throwable throwable, String message, Object... args) {
+        if (throwable != null && message != null) {
+            message += " : " + throwable.toString();
```

Comment:
Throwable toString method print is not meaningful for humans, changing it makes sense. it is same in 118th line too